compiler/src/iree/compiler/Codegen/Common/TileAndDistributeToWorkgroupsPass.cpp

```diff
@@ -148,25 +147,47 @@ static LogicalResult lowerDispatchWorkgroupCountFromDagRootOp(
   llvm::DenseSet<unsigned> partitionableLoopsSet;
   partitionableLoopsSet.insert(partitionableLoops.begin(),
                                partitionableLoops.end());
+
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(*rootOp);
+
   for (auto workload : llvm::enumerate(workloadValues)) {
     if (!partitionableLoopsSet.count(workload.index())) {
       tileSizes[workload.index()] = 0;
     }
-    if (tileSizes[workload.index()] == 0) {
+    int64_t tileSize = tileSizes[workload.index()];
+
+    if (tileSize == 0) {
       numTiles.push_back(one);
       continue;
     }
-    if (tileSizes[workload.index()] == 1) {
-      numTiles.push_back(workload.value());
-      continue;
+
+    // When the loop range is knwon to be static, let's directly use it.
+    int64_t loopRange = ShapedType::kDynamicSize;
+    if (linalgOp) {
+      loopRange = linalgOp.getStaticLoopRanges()[workload.index()];
+    }
+
+    if (loopRange != ShapedType::kDynamicSize) {
```

Comment:
One thing you can do to combine the if and the else parts is to use `makeComposedFoldedAffineApply` from upstream. If `loopRange` is dynamic use the workload, if static, then use the static value...
See if this makes sense to you. The overall logic is fine.

Reply:
Since the other values, loopRange and tileSize, are already in the form of int64_t, let keep this way, which may be simpler & faster. If the readability is of concern, I can create a lambda function for the if and else, but I don't think it will improve the readability a lot.